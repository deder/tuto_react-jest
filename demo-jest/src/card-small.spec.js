import CardSmall from './card-small';

describe("<CardSmall />",()=>{
    it("Snapshot", () => {
        const wrapper = shallow(
          <CardSmall />
        );
        expect(wrapper).to.matchSnapshot();
    });
    it("onClick", () => {
        const onClick = sinon.spy();
        const wrapper = shallow(
          <CardSmall onClick={onClick} />
        );
        wrapper.simulate('click');
        expect(onClick).to.have.been.called();
    });
})
