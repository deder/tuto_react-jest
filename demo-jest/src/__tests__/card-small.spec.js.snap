// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardSmall /> Snapshot 1`] = `
<div
  className="card col l3 s12 m12"
  style={
    Object {
      "height": "auto",
      "marginBottom": 0,
      "marginTop": 0,
      "paddingBottom": "10px",
      "paddingTop": "10px",
    }
  }
>
  <div
    className="card-image waves-effect waves-block waves-light"
    style={
      Object {
        "maxHeight": "300px",
      }
    }
  >
    <div
      className="card-content"
      style={
        Object {
          "bottom": 0,
          "height": "90px",
          "maxHeight": "none",
          "overflow": "inherit",
          "padding": "24px",
          "position": "absolute",
          "width": "100%",
        }
      }
    >
      <span
        className="card-title activator grey-text text-darken-4"
        style={
          Object {
            "color": "white !important",
            "display": "inline-block",
            "fontFamily": "Roboto",
            "fontWeight": 500,
            "lineHeight": 1.2,
            "opacity": 0.9,
            "padding": "0",
            "paddingBottom": "5px",
            "paddingLeft": "24px",
            "paddingRight": "24px",
            "textAlign": "left",
            "verticalAlign": "middle",
            "width": "100%",
            "zIndex": 2,
          }
        }
      />
      <div
        className="card-content-background"
        style={
          Object {
            "backgroundColor": "#004D40",
            "height": "100%",
            "left": 0,
            "opacity": 0.6,
            "position": "absolute",
            "top": 0,
            "width": "100%",
            "zIndex": 1,
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "background": "linear-gradient(-180deg, #004D40, rgba(255, 255, 255, 0))",
          "bottom": 0,
          "height": "100%",
          "left": 0,
          "opacity": 0.4,
          "position": "absolute",
          "width": "100%",
        }
      }
    />
    <img
      className="activator"
    />
  </div>
</div>
`;
